cpp/cmake_modules/FindgRPCAlt.cmake

```diff
@@ -24,222 +24,57 @@ if(gRPC_FOUND)
   return()
 endif()
 
-unset(GRPC_ALT_VERSION)
-
-if(ARROW_GRPC_USE_SHARED)
-  set(GRPC_GPR_LIB_NAMES)
-  set(GRPC_GRPC_LIB_NAMES)
-  set(GRPC_GRPCPP_LIB_NAMES)
-  set(GRPC_ADDRESS_SORTING_LIB_NAMES)
-  set(GRPC_UPB_LIB_NAMES)
-  if(CMAKE_IMPORT_LIBRARY_SUFFIX)
-    list(APPEND GRPC_GPR_LIB_NAMES
-                "${CMAKE_IMPORT_LIBRARY_PREFIX}gpr${CMAKE_IMPORT_LIBRARY_SUFFIX}")
-    list(APPEND GRPC_GRPC_LIB_NAMES
-                "${CMAKE_IMPORT_LIBRARY_PREFIX}grpc${CMAKE_IMPORT_LIBRARY_SUFFIX}")
-    list(APPEND GRPC_GRPCPP_LIB_NAMES
-                "${CMAKE_IMPORT_LIBRARY_PREFIX}grpc++${CMAKE_IMPORT_LIBRARY_SUFFIX}")
-    list(
-      APPEND GRPC_ADDRESS_SORTING_LIB_NAMES
-             "${CMAKE_IMPORT_LIBRARY_PREFIX}address_sorting${CMAKE_IMPORT_LIBRARY_SUFFIX}"
-      )
-    list(APPEND GRPC_UPB_LIB_NAMES
-                "${CMAKE_IMPORT_LIBRARY_PREFIX}upb${CMAKE_IMPORT_LIBRARY_SUFFIX}")
+find_package(PkgConfig QUIET)
+pkg_check_modules(GRPCPP_PC grpc++)
+if(GRPCPP_PC_FOUND)
+  set(gRPCAlt_VERSION "${GRPCPP_PC_VERSION}")
+  set(GRPCPP_INCLUDE_DIRECTORIES ${GRPCPP_PC_INCLUDEDIR})
+  if(ARROW_GRPC_USE_SHARED)
+    set(GRPCPP_LINK_LIBRARIES ${GRPCPP_PC_LINK_LIBRARIES})
+    set(GRPCPP_LINK_OPTIONS ${GRPCPP_PC_LDFLAGS_OTHER})
+    set(GRPCPP_COMPILE_OPTIONS ${GRPCPP_PC_CFLAGS_OTHER})
+  else()
+    set(GRPCPP_LINK_LIBRARIES)
+    foreach(GRPCPP_LIBRARY_NAME ${GRPCPP_PC_STATIC_LIBRARIES})
```

Comment:
Should there be an assertion here to check that the required libraries are in `GRPCPP_PC_STATIC_LIBRARIES` before iterating over them?

Reply:
I don't think that we need it.
Because we already check `GRPCC_PC_FOUND`. We can assume that `pkg_check_modules()` sets `GRPCPP_PC_*` and `GRPCPP_PC_STATIC_*` variables.
(I don't think that it can be occurred but) If `GRPCPP_PC_STATIC_LIBRARIES` is empty, the following `list(GET GRPCPP_LINK_LIBRARIES 0 GRPCPP_IMPORTED_LOCATION)` is failed. So we can detect invalid value eventually.